pkg/cmd/start/start.go

```diff
@@ -105,94 +105,8 @@ func NewGatherAndUpload() *cobra.Command {
-			klog.Exit("error: --config is required")
-		}
-		unstructured, _, _, err := cfg.Config()
-		if err != nil {
-			klog.Exit(err)
-		}
-		cont, err := config.LoadConfig(operator.Controller, unstructured.Object, config.ToDisconnectedController)
-		if err != nil {
-			klog.Exit(err)
-		}
-		operator.Controller = cont
-
-		var clientConfig *rest.Config
-		if kubeConfigPath := cmd.Flags().Lookup("kubeconfig").Value.String(); len(kubeConfigPath) > 0 {
-			kubeConfigBytes, err := os.ReadFile(kubeConfigPath) //nolint: govet
-			if err != nil {
-				klog.Exit(err)
-			}
-			kubeConfig, err := clientcmd.NewClientConfigFromBytes(kubeConfigBytes)
-			if err != nil {
-				klog.Exit(err)
-			}
-			clientConfig, err = kubeConfig.ClientConfig()
-			if err != nil {
-				klog.Exit(err)
-			}
-		} else {
-			clientConfig, err = rest.InClusterConfig()
-			if err != nil {
-				klog.Exit(err)
-			}
-		}
-		protoConfig := rest.CopyConfig(clientConfig)
-		protoConfig.AcceptContentTypes = pbAcceptContentTypes
-		protoConfig.ContentType = pbContentType
-
-		ctx, cancel := context.WithTimeout(context.Background(), operator.Interval)
-
-		// before calling the Gather logic, this command must check the featuregate status.
-		configClient, err := configv1client.NewForConfig(clientConfig)
-		if err != nil {
-			klog.Exitf("error: building client: %v", err)
-		}
-		configInformers := configv1informers.NewSharedInformerFactory(configClient, 10*time.Minute)
-
-		desiredVersion := missingVersion
-		if envVersion, exists := os.LookupEnv("RELEASE_VERSION"); exists {
-			desiredVersion = envVersion
-		}
-
-		// By default, this will exit(0) the process if the featuregates ever change to a different set of values.
-		featureGateAccessor := featuregates.NewFeatureGateAccess(
-			desiredVersion, missingVersion,
-			configInformers.Config().V1().ClusterVersions(), configInformers.Config().V1().FeatureGates(),
-			events.NewLoggingEventRecorder("insights-gather", clock.RealClock{}),
-		)
-		go featureGateAccessor.Run(ctx)
-		go configInformers.Start(ctx.Done())
-
-		select {
-		case <-featureGateAccessor.InitialFeatureGatesObserved():
-			featureGates, _ := featureGateAccessor.CurrentFeatureGates()
-			klog.Infof("FeatureGates initialized: knownFeatureGates=%v", featureGates.KnownFeatures())
-		case <-time.After(1 * time.Minute):
-			klog.Errorf("timed out waiting for FeatureGate detection")
-			klog.Exit(fmt.Errorf("timed out waiting for FeatureGate detection"))
-		}
-
-		featureGates, err := featureGateAccessor.CurrentFeatureGates()
-		if err != nil {
-			klog.Exit(err)
-		}
-		operator.InsightsConfigAPIEnabled = featureGates.Enabled(features.FeatureGateInsightsConfigAPI)
-		operator.RuntimeExtractorEnabled = featureGates.Enabled(features.FeatureGateInsightsRuntimeExtractor)
-		err = operator.Gather(ctx, clientConfig, protoConfig)
-		if err != nil {
-			klog.Error(err)
-		}
-		cancel()
-		os.Exit(0)
-	}
```

Comment:
Just for the context. My plan was to remove this when the techpreview is promoted. This corresponds to https://docs.openshift.com/container-platform/4.17/support/remote_health_monitoring/remote-health-reporting-from-restricted-network.html#insights-operator-one-time-gather_remote-health-reporting-from-restricted-network which won't be needed anymore.